core/src/main/java/org/jdbi/v3/core/transaction/LocalTransactionHandler.java

```diff
@@ -135,7 +135,7 @@ public boolean isInTransaction(Handle handle) {
             if (!didTxnRollback.get()) {
                 handle.commit();
             }
-        } catch (Exception e) {
+        } catch (Exception | Error e) {
```

Comment:
Hi @stevenschlansker,

I think this should be `Throwable`. The `Error` case is very specific and I assume it was only provided in #1405 as an example to demonstrate the problem, but in a wild there is a bunch of `Throwable` monsters (especially in Scala) which are neither `Exception` nor `Error` but which can easily flip stack trace upside down.

Therefore, if there are no other reasons to limit a range of throwables being caught here, I think this should be changed to:

```java
} catch (Throwable t) {
```

Both `Error` and `Exception` are instances of `Throwable`, but the `Error` is **not** instance of `X extends Exception`. I guess this works only because IDE or compiler did not pick it up since generics are erased in compile time and `X extends Exception` disappears.

Reply:
The `X extends Exception` approach is how we get exception transparency in Jdbi calls. That is, `inTransaction()` can only throw whatever checked exceptions are thrown by your lambda. That's how we got away from wrapping every exception in e.g. `DBIException` like we did in v2.

Reply:
From Error javadoc:

> An Error is a subclass of Throwable that indicates serious problems that a reasonable application should not try to catch.

Since we're no longer catching `Throwable` and casting to `X`, do we actually need this change?

Reply:
> The X extends Exception approach is how we get exception transparency in Jdbi calls.
(although it really ought to have been `X extends Throwable`, but we didn't figure that out until too late to change it for v3)

Reply:
Why not catch throwable instead of only the subtypes error and exception?

Reply:
I'm surprised `catch (Throwable e)` compiles.  The method `throws X (extends Exception)` and there are many throwables that don't match that signature... you must be right about funky erasure semantics here.